src/Common/src/Microsoft/Win32/SafeHandles/SafeX509Handles.Unix.cs

```diff
@@ -92,4 +92,26 @@ public override bool IsInvalid
             get { return handle == IntPtr.Zero; }
         }
     }
+
+    [SecurityCritical]
+    internal sealed class SafeSharedX509StackHandle : SafeHandle
+    {
+        internal static readonly SafeSharedX509StackHandle InvalidHandle = new SafeSharedX509StackHandle();
+
+        private SafeSharedX509StackHandle() :
+            base(IntPtr.Zero, ownsHandle: false)
+        {
+        }
+
+        protected override bool ReleaseHandle()
+        {
+            SetHandle(IntPtr.Zero);
```

Comment:
We have a SafeHandle tracking the PKCS#7 object.  Then someone wants to make use of `p7->d.signed->certs`.  We shouldn't free that value when we're done with it, because it may either cause problems with freeing the PKCS#7 later, or we merely have blocked ourselves out of using the data any longer.

So this is shared with another safehandle.

Having written out the explanation, I'm wondering if this SafeHandle needs a field reference to the owning SafeHandle (to ensure it's lifetime WRT finalization), have IsInvalid take that handle's IsInvalid into account, and ReleaseHandle just set that field to null.

Or maybe just downgrade it to IntPtr.

Since I've clearly caused confusion, I'm very open to feedback on making it easier to follow (and if it's just a comment that's certainly doable, too).